github-runner-manager/src/github_runner_manager/http_server.py

```diff
@@ -45,11 +46,13 @@ def check_runner() -> tuple[str, int]:
     Returns:
         Information on the runners in JSON format.
     """
-    app_config = app.config[APP_CONFIG_NAME]
+    app_config: ApplicationConfiguration = app.config[APP_CONFIG_NAME]
     app.logger.info("Checking runners...")
     runner_scaler = get_runner_scaler(app_config)
     try:
         runner_info = runner_scaler.get_runner_info()
+        BUSY_RUNNERS_COUNT.labels(app_config.name).set(runner_info.busy)
+        IDLE_RUNNERS_COUNT.labels(app_config.name).set(runner_info.online)
```

Comment:
couldn't we set those in reconcile? `check_runner` is only triggered by a manual action.

Reply:
get_runner_info is only called from here and nowhere else

Reply:
Then doesn't that also go against the value of having side effects as the previous comment?

Reply:
I'm not sure if I understand you correctly, but I would expect a method called `_issue_reconciliation_metric` to have side effects. However, I would expect a method like `get_runner_info` to simply return a value without side effects or changing the state.